File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/model/base/LocManager.java
package com.example.ama.android2_lesson03.model.base;

import android.location.Address;

import com.google.android.gms.maps.model.LatLng;

public interface LocManager {
    Address findAddressByQuery(String query);

    Address findAddressByCoords(LatLng coords);

    void findMyLocation(QueryManager.OnLocationSearchResultCallback callback);


}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/repo/data/LocManager.kt
package com.example.ama.android2_lesson03.repo.data

import android.location.Address
import com.example.ama.android2_lesson03.repo.base.QueryManager
import com.google.android.gms.maps.model.LatLng

interface LocManager {
    fun findAddressByQuery(query: String): Address
    fun findAddressByLatLng(latLng: LatLng): Address
    fun fingMyLocation(callback: QueryManager.OnLocationSearchResultCallback)
}